Add guideline for removal of irrelevant data (#6018)
mdn/browser-compat-data: docs/data-guidelines.md

For Safari for iOS, use the iOS version number, not the Safari version number or WebKit version number.

This versioning scheme came at [Apple's request, in #2006](https://github.com/mdn/browser-compat-data/issues/2006#issuecomment-457277312).

> ## Removal of irrelevant data

Reviewer: A suggestion to retitle this. I think we have different standards for browser data, so I think it would be good to narrow the scope a little bit.

```suggestion
## Removal of irrelevant features
```

> Data can be removed from BCD if it is considered irrelevant. This can happen if:

Reviewer: ```suggestion
Features can be removed from BCD if it is considered irrelevant. A feature can be considered irrelevant if any of these conditions are met:
```

> - a feature isn't implemented in any browser and the specification has been abondoned.

Reviewer: Reworded this to distinguish more clearly from the second condition. If this is not what you meant, please say so. 😄 

```suggestion
- a feature was never implemented in any browser and the specification has been abandoned.
```

> - a feature has been removed from all browsers and wasn't present in releases dating back 2 years.

Reviewer: ```suggestion
- a feature was implemented and has since been removed from all browser releases dating back 2 years.
```
